### ISSUE TITLE
[ECS] [request]: Support for restricting containers to a single NUMA node

### ISSUE THREAD

**Issue by tomelliff** [aws/containers-roadmap#1093]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

**Tell us about your request**
Docker allows us to specify `cpuset-cpus` to pin to specific CPUs or more widely to use `cpuset-mems` to pin to specific NUMA nodes and prevent the container trying to use memory across another socket which can cause massive performance hits. I'd like this to be exposable by the ECS API but also to have a higher level requirement where we can just state the number of NUMA nodes it can use (which we would like to stick as 1). This higher level feature would then mean it's much easier to just throw any CPU heavy workload at a dual socket (or more) instance and not have to worry about NUMA issues.

**Which service(s) is this request for?**
ECS

**Tell us about the problem you're trying to solve. What are you trying to do, and why is it hard?**
We have some CPU heavy tasks that require a lot of memory access. When we run them on a c5.18xl or c5.24xl instance then they are much slower than when we run on c5.9xl or c5.12xl instances due to NUMA issues where computation is limited by memory access times when going across cores.

**Are you currently working around this issue?**
We're using single socket nodes only. We'd like to be able to be flexible and use autoscaling groups that support mixed instance types including the dual socket nodes.

**Comment by scottmalkie** [comment]
Tangentially related, but AWS Graviton2-based EC2 instances (arm64) are fully supported by ECS and have no NUMA issues.  See [here](https://github.com/aws/aws-graviton-getting-started) and [here](https://aws.amazon.com/blogs/compute/c6g-openfoam-better-price-performance/) for more information.

**Comment by tomelliff** [comment]
Yep and that's definitely tempting for us. We've managed to run our main workloads on the m6gs when they were first released but haven't made a decision yet on whether we can run on them in production normally due to the impact it has on debugging when things go wrong. Would be nice if our devs were also running ARM64 chips locally so they could then still easily pull the production deployed container and run it locally to debug things.
